### PR TITLE
Implemented some hostgroup tests based on known bugs

### LINKED ISSUE #3591: Hammer: Add more tests around hostgroup cmd
Currently automation has multiple tests but most of them are implemented with only one option with hostgroup cmd.
We should have tests where hostgroup cmd should be created with multiple options like:

````
 ~]# hammer -u admin -p changeme hostgroup create --name test123512 --organizations='Default Organization' --location-ids 1 --puppet-proxy-id 1   --puppet-ca-proxy-id 1 --root-pass redhat123 --subnet-id 12 --domain-id 1 --architecture-id 1
Could not create the hostgroup:
  resource have no errors
````

When subnet-id is incorrect, user should get proper error:
````
~]# hammer -u admin -p changeme hostgroup create --name test123512 --organizations='Default Organization' --location-ids 1 --puppet-proxy-id 1   --puppet-ca-proxy-id 1 --root-pass redhat123 --subnet-id 1 --domain-id 1
Could not create the hostgroup:
  ERROR:  insert or update on table "hostgroups" violates foreign key constraint "hostgroups_subnet_id_fk"
  DETAIL:  Key (subnet_id)=(1) is not present in table "subnets".
```
With arch name, following error raised, which is misleading:
```
 ~]# hammer -u admin -p changeme hostgroup create --name test12342 --architecture=x86_64 --organizations='Default Organization' 
Could not create the hostgroup:
  Error: found more than one architecture

~]# hammer -u admin -p changeme architecture list
---|-------
ID | NAME  
---|-------
2  | i386  
1  | x86_64
---|-------
[~]# hammer -u admin -p changeme hostgroup create --name test12342 --architecture-id=1 --organizations='Default Organization' 
Hostgroup created
```

Also, we should have hostgroup tests with all possible cmd line options.

### PR DESCRIPTION
**Pull request by sghai**
Closes #3591 

I see some issues when creating hostgroup via hammer. So based on those issues, new tests have been added. They are currently blocked by bz's.

### CODE REVIEW

**Comment by sghai** [comment]
test results:

```
 robottelo]$ py.test ./tests/foreman/cli/test_hostgroup.py::HostGroupTestCase================================================================ test session starts =================================================================
platform linux2 -- Python 2.7.5, pytest-2.9.0, py-1.4.31, pluggy-0.3.1
rootdir: /home/sghai/QE/robottelo, inifile: 
collected 19 items 

tests/foreman/cli/test_hostgroup.py ss.s..s..s.s.......

======================================================= 13 passed, 6 skipped in 693.38 seconds =======================================================

```

**Comment by rochacbruno** [comment]
ACK

**Comment by sghai** [comment]
Rebased.

**Comment by sghai** [comment]
Guys, need one more reviewer :)

File: tests/foreman/cli/test_hostgroup.py [hunk 1]
```diff
@@ -151,6 +157,173 @@ def test_positive_create_with_puppet_proxy(self):
...
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_create_with_domain(self):
+        """Check if hostgroup with domain can be created
+
+        @id: c468fcac-9e42-4ee6-a431-abe29b6848ce
+
+        @Assert: Hostgroup should be created and has domain assigned
+
```
Comment (oshtaier) [inline tests/foreman/cli/test_hostgroup.py]: Redundant line

File: tests/foreman/cli/test_hostgroup.py [hunk 2]
```diff
@@ -151,6 +157,173 @@ def test_positive_create_with_puppet_proxy(self):
...
+        @id: 21c619f4-7339-4fb0-9e29-e12dae65f943
+
+        @Assert: Hostgroup should be created and has architecture assigned
+
+        @BZ: 1354544
+        """
+        arch = 'x86_64'
+        hostgroup = make_hostgroup({'architecture': arch})
+        self.assertEqual(
```
Comment (oshtaier) [inline tests/foreman/cli/test_hostgroup.py]: Can we write that in one line?
```
+        self.assertEqual(
 +            arch,
 +            hostgroup['architecture'],
 +        )
```

File: tests/foreman/cli/test_hostgroup.py [hunk 3]
```diff
@@ -151,6 +157,173 @@ def test_positive_create_with_puppet_proxy(self):
...
+        """Check if hostgroup with domain can be created
+
+        @id: c468fcac-9e42-4ee6-a431-abe29b6848ce
+
+        @Assert: Hostgroup should be created and has domain assigned
+
+        """
+        domain = make_domain()
+        hostgroup = make_hostgroup({'domain-id': domain['id']})
+        self.assertEqual(
```
Comment (oshtaier) [inline tests/foreman/cli/test_hostgroup.py]: Same here, I feel that statement should take less than 80 characters

File: tests/foreman/cli/test_hostgroup.py [hunk 4]
```diff
@@ -151,6 +157,173 @@ def test_positive_create_with_puppet_proxy(self):
...
+        domain = make_domain()
+        org = make_org()
+        lc_env = make_lifecycle_environment({'organization-id': org['id']})
+        loc = make_location()
+        env = make_environment()
+        os = make_os()
+        media = make_medium()
+        arch = make_architecture()
+        puppet_proxy = Proxy.list()[0]
+        ptable = 'Kickstart default'
```
Comment (oshtaier) [inline tests/foreman/cli/test_hostgroup.py]: Maybe we can create ptable too

File: tests/foreman/cli/test_hostgroup.py [hunk 5]
```diff
@@ -151,6 +157,173 @@ def test_positive_create_with_puppet_proxy(self):
...
+            'puppet-proxy-id': puppet_proxy['id'],
+            'puppet-ca-proxy-id': puppet_proxy['id'],
+            'operatingsystem-id': os['id'],
+            'architecture-id': arch['id'],
+            'lifecycle-environment': lc_env['name'],
+            'lifecyle-environment-organization-id': org['id'],
+            'partition-table': ptable,
+            'medium-id': media['id'],
+        })
+        self.assertEqual(domain['name'], hostgroup['domain'])
```
Comment (oshtaier) [inline tests/foreman/cli/test_hostgroup.py]: You created so many entities, but have only 4 assertions, I would like the idea to have same amounts of entities and assertions whether less of one or more of last

File: tests/foreman/cli/test_hostgroup.py [hunk 6]
```diff
@@ -151,6 +157,173 @@ def test_positive_create_with_puppet_proxy(self):
...
+
+        @BZ: 1354568
+        """
+        subnet_id = gen_string('numeric', 4)
+        with self.assertRaises(CLIReturnCodeError) as exception:
+            HostGroup.create({
+                'name': gen_string('alpha', 10),
+                'subnet-id': subnet_id
+            })
+        self.assertIs(exception.exception.stderr, 'Could not find subnet {0}'
```
Comment (oshtaier) [inline tests/foreman/cli/test_hostgroup.py]: self.assertIs(
    exception.exception.stderr, 
    'Could not find subnet {0}'.format(subnet_id)
)

File: tests/foreman/cli/test_hostgroup.py [hunk 7]
```diff
@@ -151,6 +157,173 @@ def test_positive_create_with_puppet_proxy(self):
...
+        @id: c352d7ea-4fc6-4b78-863d-d3ee4c0ad439
+
+        @Assert: Proper error should be raised
+
+        @BZ: 1354568
+        """
+        subnet_id = gen_string('numeric', 4)
+        with self.assertRaises(CLIReturnCodeError) as exception:
+            HostGroup.create({
+                'name': gen_string('alpha', 10),
```
Comment (oshtaier) [inline tests/foreman/cli/test_hostgroup.py]: `gen_string('alpha')` as it is already 10 by default

File: tests/foreman/cli/test_hostgroup.py [hunk 8]
```diff
@@ -151,6 +157,173 @@ def test_positive_create_with_puppet_proxy(self):
...
+        @id: b36c83d6-b27c-4f1a-ac45-6c4999005bf7
+
+        @Assert: Proper error should be raised
+
+        @BZ: 1354568
+        """
+        domain_id = gen_string('numeric', 4)
+        with self.assertRaises(CLIReturnCodeError) as exception:
+            HostGroup.create({
+                'name': gen_string('alpha', 10),
```
Comment (oshtaier) [inline tests/foreman/cli/test_hostgroup.py]: same here about length

File: tests/foreman/cli/test_hostgroup.py [hunk 9]
```diff
@@ -151,6 +157,173 @@ def test_positive_create_with_puppet_proxy(self):
...
+                'domain-id': domain_id
+            })
+        self.assertIs(exception.exception.stderr, 'Could not find domain {0}'
```
Comment (oshtaier) [inline tests/foreman/cli/test_hostgroup.py]: same here about format

File: tests/foreman/cli/test_hostgroup.py [hunk 10]
```diff
@@ -151,6 +157,173 @@ def test_positive_create_with_puppet_proxy(self):
...
+        @id: 7b7de0fa-aee9-4163-adc2-354c1e720d90
+
+        @Assert: Proper error should be raised
+
+        @BZ: 1354568
+        """
+        arch_id = gen_string('numeric', 4)
+        with self.assertRaises(CLIReturnCodeError) as exception:
+            HostGroup.create({
+                'name': gen_string('alpha', 10),
```
Comment (oshtaier) [inline tests/foreman/cli/test_hostgroup.py]: there

File: tests/foreman/cli/test_hostgroup.py [hunk 11]
```diff
@@ -151,6 +157,173 @@ def test_positive_create_with_puppet_proxy(self):
...
+                'architecture-id': arch_id
+            })
+        self.assertIs(exception.exception.stderr,
```
Comment (oshtaier) [inline tests/foreman/cli/test_hostgroup.py]: and there

**Comment by oshtaier** [comment]
ACK pending comments (most of them, of course, non-blockers)

**Comment by sghai** [comment]
@oshtaier : thank you for all review comments. All comments are addressed. Please take a another look. thanks

**Comment by oshtaier** [comment]
ACK

**Pull request merged**